atc/atccmd/command.go

```diff
@@ -1171,6 +1175,13 @@ func (cmd *RunCommand) validate() error {
 				errors.New("must specify HTTPS external-url to use TLS"),
 			)
 		}
+	case cmd.TLSOnly:
+		if !cmd.isTLSEnabled() || !cmd.LetsEncrypt.Enable {
+			errs = multierror.Append(
+				errs,
+				errors.New("must specify a --tls-bind-port, or --enable-lets-encrypt to use TLS "),
```

Comment:
This won't work because the `switch` statement run the `cmd.TLSBindPort == 0` (line 1157) instead. In fact it seems like a lot of these validations are kind of awkwardly overlapping as a result of using `switch`. :thinking: It's really one big chained `else-if`.

Using a `switch` statement here feels like it's not aging well. It could be refactored, or you could just put a `if cmd.TLSOnly` in the `cmd.TLSBindPort == 0` branch. I'd be fine with both; the latter is probably less work.

The error can then read `must specify --tls-bind-port in order to configure --tls-only`. I don't think you need to mention Let's Encrypt at all.